packages/echo-core/src/hooks/usePanels.ts

```diff
@@ -53,20 +53,33 @@ export function usePanels(panelType = String(PanelType.left)): UsePanels {
                     activePanelState: { ...s.app.activePanelState, ...newActivePanel }
                 }
             }));
+            setPanelUI({});
         },
         [activePanel]
     );
 
     useEffect(() => {
-        if (ui) {
-            setPanelUI({ ...ui });
-        } else {
+        if (activePanel === '') {
             setPanelUI({});
+            return;
         }
-    }, [ui]);
+        const currentActivePanel: Panel | undefined = modulePanels.find((panel: Panel) => panel.key === activePanel);
+        if (currentActivePanel && currentActivePanel.customUiStates) {
+            const activeCustomUiState =
+                currentActivePanel.defaultActiveUiState && !isPanelActive
+                    ? currentActivePanel.defaultActiveUiState
+                    : currentActivePanel?.activeCustomUiState && currentActivePanel?.activeCustomUiState !== ''
```

Comment:
unnecessary use of ?, at this point currentActivePanel can not be null/undefined